### PR TITLE
FEATURE: TNT-1407 Single selection variant of Attr. filter

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: TNT-1407
- Single selection specific subcomponents
- Default components as getter
- styling
- first storybooks

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended check plugins`                                | Dashboard plugins tests                                    |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11952/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13950/ : SUCCESS in 12m 59s

File: libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/StatusBar/SingleSelectionAttributeFilterStatusBar.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// (C) 2021-2023 GoodData Corporation
+import { IAttributeElement } from "@gooddata/sdk-model";
+import React from "react";
+import { AttributeFilterFilteredStatus } from "./AttributeFilterFilteredStatus";
+
+/**
+ * It represents component that display status of current selection.
+ * @beta
+ */
+export interface IAttributeFilterStatusBarProps {
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/StatusBar/SingleSelectionAttributeFilterStatusBar.tsx:10]: I think we already have this type somewhere else?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11996/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13981/ : FAILURE in 7m 34s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11998/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13983/ : SUCCESS in 13m 13s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1985/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11999/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/162/ : UNSTABLE in 21m 50s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12006/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13990/ : SUCCESS in 14m 07s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1987/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12007/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/164/ : UNSTABLE in 27m 53s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12011/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13993/ : SUCCESS in 13m 15s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12012/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/165/ : SUCCESS in 22m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12015/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13997/ : SUCCESS in 13m 40s

File: libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    // Modify item click behavior to select only this particular item.
+    const onItemClick = useCallback(
+        (event: React.MouseEvent) => {
+            event.stopPropagation();
+            onSelectOnly();
+        },
+        [onSelectOnly],
+    );
+
+    const itemTitle = getElementTitle(item);
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx:31]: Pls memo it

File: libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * Renders elements selection list item as a single select list item.
+ *
+ * @beta
+ */
+export const SingleSelectionAttributeFilterElementsSelectItem: React.VFC<
+    IAttributeFilterElementsSelectItemProps
+> = (props) => {
+    const { item, onSelectOnly, isSelected, fullscreenOnMobile = false } = props;
+
+    // Modify item click behavior to select only this particular item.
```
Comment (ivanmjartan) [inline libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx:22]: What does mean this comment ?

**Comment by ivanmjartan** [comment]
I am missing basic e2e for new Attribute filer mode in sdk-ui-tests-e2e (Cypress). 

Happy path, Parent filter relation, Mutli vs Single select conversion.

File: common/changes/@gooddata/sdk-ui-all/ine-tnt-1407_2023-03-17-15-15.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "changes": [
+        {
+            "packageName": "@gooddata/sdk-ui-all",
+            "comment": "Single selection variant of Attribute Filter component",
+            "type": "none"
+        }
+    ],
+    "packageName": "@gooddata/sdk-ui-all"
+}
```
Comment (ivanmjartan) [inline common/changes/@gooddata/sdk-ui-all/ine-tnt-1407_2023-03-17-15-15.json:10]: I am missing basic e2e for new Attribute filer mode in sdk-ui-tests-e2e (Cypress).

Happy path, Parent filter relation, Mutli vs Single select conversion.

**Comment by ivanmjartan** [comment]
extended test - cypress - isolated

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**drilling.spec.ts**|7(2 skipped)|-|37.3410s|
|✅|**dashboardDependentFilter.spec.ts**|8|-|83.8660s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|35.5860s|
|✅|**pivotTableSizing.spec.ts**|1|-|3.8590s|
|✅|**dateFilter.spec.ts**|9|-|11.6770s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|5.7700s|
|✅|**headlineChartDrilling.spec.ts**|2|-|2.7810s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|12.5780s|
|✅|**dashboardDateRelated.spec.ts**|8|-|61.0490s|
|✅|**attributeFilterButton.spec.ts**|13|-|47.2480s|
|✅|**dashboard.spec.ts**|9|-|55.9370s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.0520s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|15.6850s|
|✅|**dashboardKPI.spec.ts**|8|-|132.2190s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|135.8670s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|17.5710s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|5.7200s|
|✅|**barChartDrilling.spec.ts**|4|-|5.3740s|
|✅|**dashboardDnD.spec.ts**|7|-|54.8890s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|5.7710s|
|✅|**attributeFilter.spec.ts**|1|-|6.2010s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/455/).

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**drilling.spec.ts**|7(2 skipped)|-|37.3410s|
|✅|**dashboardDependentFilter.spec.ts**|8|-|83.8660s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|35.5860s|
|✅|**pivotTableSizing.spec.ts**|1|-|3.8590s|
|✅|**dateFilter.spec.ts**|9|-|11.6770s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|5.7700s|
|✅|**headlineChartDrilling.spec.ts**|2|-|2.7810s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|12.5780s|
|✅|**dashboardDateRelated.spec.ts**|8|-|61.0490s|
|✅|**attributeFilterButton.spec.ts**|13|-|47.2480s|
|✅|**dashboard.spec.ts**|9|-|55.9370s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.0520s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|15.6850s|
|✅|**dashboardKPI.spec.ts**|8|-|132.2190s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|135.8670s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|17.5710s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|5.7200s|
|✅|**barChartDrilling.spec.ts**|4|-|5.3740s|
|✅|**dashboardDnD.spec.ts**|7|-|54.8890s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|5.7710s|
|✅|**attributeFilter.spec.ts**|1|-|6.2010s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/455/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12022/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/170/ : SUCCESS in 20m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12023/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14002/ : SUCCESS in 13m 18s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivanmjartan** [comment]
extended test - tiger-cypress - isolated

File: libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx (hunk 3 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/SingleSelectionAttributeFilterElementsSelectItem.tsx:20]: description of difference from "default" behavior

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12026/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/171/ : FAILURE in 10m 51s

File: common/changes/@gooddata/sdk-ui-all/ine-tnt-1407_2023-03-17-15-15.json (hunk 4 above)
Comment (ivan-nejezchleb) [reply common/changes/@gooddata/sdk-ui-all/ine-tnt-1407_2023-03-17-15-15.json:10]: Will be delivered by follow up PR

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3553

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboardDateRelated.spec.ts**|8|-|62.7870s|
|✅|**dateFilter.spec.ts**|9|-|11.7110s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|5.8280s|
|✅|**dashboardFiltering.spec.ts**|18(2 skipped)|-|137.1840s|
|✅|**pivotTableSizing.spec.ts**|1|-|3.8070s|
|✅|**dashboardKPI.spec.ts**|8|-|133.3800s|
|✅|**dashboard.spec.ts**|9|-|55.2060s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|12.5460s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.0370s|
|✅|**headlineChartDrilling.spec.ts**|2|-|2.9200s|
|✅|**attributeFilter.spec.ts**|1|-|5.7190s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|15.7160s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|5.9070s|
|✅|**dashboardDependentFilter.spec.ts**|8|-|85.7540s|
|✅|**dashboardDnD.spec.ts**|7|-|55.9770s|
|✅|**barChartDrilling.spec.ts**|4|-|5.3900s|
|✅|**attributeFilterButton.spec.ts**|13|-|47.0370s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|17.7100s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|35.8620s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|5.7910s|
|✅|**drilling.spec.ts**|7(2 skipped)|-|37.1230s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/456/).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul/79/ : FAILURE in 12m 26s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/409/ : FAILURE in 1m 47s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14006/ : SUCCESS in 15m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3682/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2525/ : SUCCESS in 11m 03s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul/456/ : SUCCESS in 20m 38s (non-voting)

**Pull request merged**